automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UISpec.scala

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.dsde.firecloud.fixture
+
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.{Credentials, UserPool}
+import org.broadinstitute.dsde.workbench.fixture._
+import org.broadinstitute.dsde.workbench.service.test.{RandomUtil, WebBrowserSpec}
+import org.scalatest.{BeforeAndAfterAll, Matchers, Outcome, fixture}
+
+import scala.util.Try
+
+
+abstract class UISpec extends fixture.FreeSpec with BeforeAndAfterAll with BillingFixtures with Matchers
+  with WebBrowserSpec with UserFixtures with WorkspaceFixtures with GroupFixtures with RandomUtil {
+
+  val owner: Credentials = UserPool.chooseProjectOwner
+  implicit val ownerAuthToken: AuthToken = owner.makeAuthToken()
+  var claimedBillingProject: ClaimedProject = _
+
+  /**
+    * See
+    *  https://www.artima.com/docs-scalatest-2.0.M5/org/scalatest/FreeSpec.html
+    *   Section: "Overriding withFixture(OneArgTest)"
+    *
+    * @param billingProject
+    */
+  case class projectOwnerFixture(billingProject: String)
+
+  type FixtureParam = projectOwnerFixture
+
+  override def withFixture(test: OneArgTest): Outcome = {
+    withFixture(test.toNoArgTest(projectOwnerFixture(claimedBillingProject.projectName)))
+  }
+
+  override def beforeAll(): Unit = {
```

Comment:
Using `BeforeAndAfterAll` changes the lifecycle of the claimed project. Maybe it's not a problem for `DataSpec` but I want to make sure that you considered that.

It's also potentially inconvenient that tests do not have control over the lifecycle. An alternative would be for this abstract class to define an abstract function for getting the billing project to provide to the test case. That way the extending suite can choose whether to use `BeforeAndAfter`/`BeforeAndAfterEach`/`BeforeAndAfterAll`.